# Q.1 A F&B manager wants to determine whether there is     any significant difference in the diameter of the     cutlet between two units. A randomly selected         sample of cutlets was collected from both units       and measured? Analyze the data and draw inferences     at 5% significance level. Please state the             assumptions and tests that you carried out to         check validity of the assumptions.

Ans :
Step 1 : Define the Hypothesis.

Null Hypothesis(H0): µ1 = µ2 (There is no significant difference in the diameter of the cutlet between two units.)

Alternate Hypothesis(Ha): µ1 ≠ µ2 (There is significant difference in the diameter of the cutlet between two units.)

Step 2 : Decide which test to be done.

We'll conduct t-test.


Step 3 : Get the α value

Significance = 5%

α = 0.05


Step 4 : Identify whether the test is one tailed or two tailed.

As we are considering two possibilities,  it is a two-tailed test.

α/2 = 0.025


In [21]:
# Import Libraries
import pandas as pd 
import numpy as np
import scipy.stats as stats
import statsmodels.api as sm 

In [3]:
cutlets = pd.read_csv('Cutlets.csv')
cutlets

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

Step 5 : Calculate p_value

In [19]:
statistic, p_value = stats.ttest_ind(cutlets['Unit A'], cutlets['Unit B'], alternative='two-sided')
print('p_value = ', p_value)

p_value =  0.4722394724599501


Step 6 : Compare p_value with α

In [20]:
α = 0.05
print('Significance=%.3f, p=%.3f' %(α ,p_value))
if p_value <= α:
    print('We reject Null Hypothesis.')
else:
    print('We fail to reject Null Hypothesis.')

Significance=0.050, p=0.472
We fail to reject Null Hypothesis.


Hence, we fail to reject null hypothesis

So there is no significant difference in the diameter of the cutlet between two units

# Q.2 A hospital wants to determine whether there is any difference in the average Turn Around Time(TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.

# Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

Ans : 

Step 1 : Define the hypothesis.

H0 : µ1 = µ2 = µ3 = µ4 (There is no difference in average TAT of reports of the laboratories.)

Ha : Atleast one of them is different.

Step 2 : Decide the test.

Here we have four samples so we'll use Anova test.

Step 3 : Get the α value

Significance = 5%

α = 0.05


In [8]:
ltat = pd.read_csv('LabTAT.csv')
ltat

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

Step 4 : Calculate p_value

In [16]:
test_statistic, p_value = stats.f_oneway(ltat.iloc[:,0],ltat.iloc[:,1],
                                         ltat.iloc[:,2],ltat.iloc[:,3],axis=0)
print('p_value =',p_value)

p_value = 2.1156708949992414e-57


Step 5 : Compare p_value with α

In [18]:
α = 0.05
print('Significance=%.3f, p=%.3f' %(α ,p_value))
if p_value <= α:
    print('We reject Null Hypothesis.')
else:
    print('We fail to reject Null Hypothesis.')

Significance=0.050, p=0.000
We reject Null Hypothesis.


Hence, we reject null hypothesis.

So, Atleast one of them is different.

# Q.3 Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [22]:
buyer = pd.read_csv('BuyerRatio.csv')
buyer

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [2]:
table = [[50,142,131,70],[435,1523,1356,750]]
table

[[50, 142, 131, 70], [435, 1523, 1356, 750]]

Step 1 : Define the hypothesis

H0 : There is no dependency between gender based byuer ratio across the regions

Ha : There is significant dependency between gender based byuer ratio across the regions



Step 2 : Decide the test

Here we'll conduct chi-square test


In [7]:

from scipy.stats import chi2_contingency

In [8]:
# Chi2 contengency independence test
chi2_contingency(table)

 # o/p is (Chi2 stats value, p_value, df, expected obsvations)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

Step 3 : Compare p_value with α = 0.05

As p_value = 0.66 is greater than α = 0.05

Hence, we fail to reject null hypothesis

i.e. There is no dependency between gender based byuer ratio across the regions

# Q.4 TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [22]:
df = pd.read_csv('Costomer+OrderForm.csv')
df

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [23]:
df.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [24]:
df.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [25]:
df.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [26]:
df.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [27]:
# Make a contingency table
table=np.array([[271,267,269,280],[29,33,31,20]])
table

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

Step 1 : Define the hypothesis

Ho: Independence of categorical variables (customer order forms defective % does not varies by centre)

Ha: Dependence of categorical variables (customer order forms defective % varies by centre)

Step 2 : Decide the test

Here we'll conduct chi-square test


In [28]:
# Chi2 contengency independence test
chi2_contingency(table)
# o/p is (Chi2 stats value, p_value, df, expected obsvations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

 Step 3 : Compare p_value with α = 0.05

As p_value = 0.2771 is greater than α = 0.05

Hence, we fail to reject null hypothesis

i.e. Independence of categorical variables Thus, 
customer order forms defective % does not varies by centre